diff --git fftools/opt_common.c fftools/opt_common.c
index ae5e28a..609dee8 100644
--- fftools/opt_common.c
+++ fftools/opt_common.c
@@ -196,6 +196,9 @@ static void print_all_libs_info(int flags, int level)
 
 static void print_program_info(int flags, int level)
 {
+    int year;
+    int month;
+    int day;
     const char *indent = flags & INDENT? "  " : "";
 
     av_log(NULL, level, "%s version " FFMPEG_VERSION, program_name);
@@ -203,8 +206,36 @@ static void print_program_info(int flags, int level)
         av_log(NULL, level, " Copyright (c) %d-%d the FFmpeg developers",
                program_birth_year, CONFIG_THIS_YEAR);
     av_log(NULL, level, "\n");
-    av_log(NULL, level, "%sbuilt with %s\n", indent, CC_IDENT);
 
+#define BUILDTM_YEAR (\
+    __DATE__[7] == '?' ? 1900 \
+    : (((__DATE__[7] - '0') * 1000 ) \
+    + (__DATE__[8] - '0') * 100 \
+    + (__DATE__[9] - '0') * 10 \
+    + __DATE__[10] - '0'))
+
+#define BUILDTM_MONTH (\
+    __DATE__ [2] == '?' ? 1 \
+    : __DATE__ [2] == 'n' ? (__DATE__ [1] == 'a' ? 1 : 6) \
+    : __DATE__ [2] == 'b' ? 2 \
+    : __DATE__ [2] == 'r' ? (__DATE__ [0] == 'M' ? 3 : 4) \
+    : __DATE__ [2] == 'y' ? 5 \
+    : __DATE__ [2] == 'l' ? 7 \
+    : __DATE__ [2] == 'g' ? 8 \
+    : __DATE__ [2] == 'p' ? 9 \
+    : __DATE__ [2] == 't' ? 10 \
+    : __DATE__ [2] == 'v' ? 11 \
+    : 12)
+
+#define BUILDTM_DAY (\
+    __DATE__[4] == '?' ? 1 \
+    : ((__DATE__[4] == ' ' ? 0 : \
+    ((__DATE__[4] - '0') * 10)) + __DATE__[5] - '0'))
+
+    year = BUILDTM_YEAR;
+    month = BUILDTM_MONTH;
+    day = BUILDTM_DAY;
+    av_log(NULL, level, "%sbuilt with '%s' on %d-%02d-%02d\n", indent, CC_IDENT, year, month, day);
     av_log(NULL, level, "%sconfiguration: " FFMPEG_CONFIGURATION "\n", indent);
 }
 
