diff --git fftools/cmdutils.c fftools/cmdutils.c
index b8de319..a0837f4 100644
--- fftools/cmdutils.c
+++ fftools/cmdutils.c
@@ -1141,12 +1141,16 @@ static void print_program_info(int flags, int level)
 {
     const char *indent = flags & INDENT? "  " : "";
 
+    char today[100];
+    time_t now = time (0);
+    strftime (today, 100, "%Y%m%d", localtime (&now));
+
     av_log(NULL, level, "%s version " FFMPEG_VERSION, program_name);
     if (flags & SHOW_COPYRIGHT)
         av_log(NULL, level, " Copyright (c) %d-%d the FFmpeg developers",
                program_birth_year, CONFIG_THIS_YEAR);
     av_log(NULL, level, "\n");
-    av_log(NULL, level, "%sbuilt with %s\n", indent, CC_IDENT);
+    av_log(NULL, level, "%sbuilt with '%s' on %s\n", indent, CC_IDENT, today);
 
     av_log(NULL, level, "%sconfiguration: " FFMPEG_CONFIGURATION "\n", indent);
 }
