diff --git a/libavformat/movenc.c b/libavformat/movenc.c
index ae49582..15fe9c6 100644
--- a/libavformat/movenc.c
+++ b/libavformat/movenc.c
@@ -6616,7 +6616,9 @@ int ff_mov_write_packet(AVFormatContext *s, AVPacket *pkt)
             av_log(s, AV_LOG_ERROR, "Malformed AAC bitstream detected: "
                    "use the audio bitstream filter 'aac_adtstoasc' to fix it "
                    "('-bsf:a aac_adtstoasc' option with ffmpeg)\n");
+            /* edited: not fatal anymore
             return -1;
+            */
         }
         av_log(s, AV_LOG_WARNING, "aac bitstream error\n");
     }
