diff --git a/ffmpeg.c b/ffmpeg.c
index 25001f5..2850746 100644
--- a/ffmpeg.c
+++ b/ffmpeg.c
@@ -895,7 +895,7 @@ static void do_video_out(AVFormatContext *s,
     nb_frames = FFMIN(nb_frames, ost->max_frames - ost->frame_number);
     if (nb_frames == 0) {
         nb_frames_drop++;
-        av_log(NULL, AV_LOG_WARNING,
+        av_log(NULL, AV_LOG_VERBOSE,
                "*** dropping frame %d from stream %d at ts %"PRId64"\n",
                ost->frame_number, ost->st->index, in_picture->pts);
         return;
