diff --git a/ffmpeg.c b/ffmpeg.c
index 25001f5..a4d97f3 100644
--- a/ffmpeg.c
+++ b/ffmpeg.c
@@ -893,13 +893,7 @@ static void do_video_out(AVFormatContext *s,
     }
 
     nb_frames = FFMIN(nb_frames, ost->max_frames - ost->frame_number);
-    if (nb_frames == 0) {
-        nb_frames_drop++;
-        av_log(NULL, AV_LOG_WARNING,
-               "*** dropping frame %d from stream %d at ts %"PRId64"\n",
-               ost->frame_number, ost->st->index, in_picture->pts);
-        return;
-    } else if (nb_frames > 1) {
+    if (nb_frames > 1) {
         if (nb_frames > dts_error_threshold * 30) {
             av_log(NULL, AV_LOG_ERROR, "%d frame duplication too large, skipping\n", nb_frames - 1);
             nb_frames_drop++;
