diff --git a/libavcodec/utils.c b/libavcodec/utils.c
index 6071380..3de6c81 100644
--- a/libavcodec/utils.c
+++ b/libavcodec/utils.c
@@ -2664,8 +2664,6 @@ int avcodec_decode_subtitle2(AVCodecContext *avctx, AVSubtitle *sub,
                     av_log(avctx, AV_LOG_ERROR,
                            "Invalid UTF-8 in decoded subtitles text; "
                            "maybe missing -sub_charenc option\n");
-                    avsubtitle_free(sub);
-                    return AVERROR_INVALIDDATA;
                 }
             }
 
