diff --git a/libavcodec/decode.c b/libavcodec/decode.c
index 4607e9f318..b50f65a86c 100644
--- a/libavcodec/decode.c
+++ b/libavcodec/decode.c
@@ -1093,9 +1093,6 @@ int avcodec_decode_subtitle2(AVCodecContext *avctx, AVSubtitle *sub,
                     av_log(avctx, AV_LOG_ERROR,
                            "Invalid UTF-8 in decoded subtitles text; "
                            "maybe missing -sub_charenc option\n");
-                    avsubtitle_free(sub);
-                    ret = AVERROR_INVALIDDATA;
-                    break;
                 }
             }
 
