diff --git a/libswscale/swscale.c b/libswscale/swscale.c
index 2795429..31e869b 100644
--- a/libswscale/swscale.c
+++ b/libswscale/swscale.c
@@ -954,10 +954,12 @@ static int scale_internal(SwsContext *c,
     if (frame_start && !scale_dst) {
         if (srcSliceY != 0 && srcSliceY + srcSliceH != c->srcH) {
             av_log(c, AV_LOG_ERROR, "Slices start in the middle!\n");
+            /* edited, removed:
             return AVERROR(EINVAL);
+            */
+            c->sliceDir = 1;
         }
-
-        c->sliceDir = (srcSliceY == 0) ? 1 : -1;
+        else c->sliceDir = (srcSliceY == 0) ? 1 : -1;
     } else if (scale_dst)
         c->sliceDir = 1;
 
